### ISSUE TITLE
Git ability to clean untracked and ignored files

### ISSUE THREAD

**Issue by benjifisher** [ansible/ansible#20554]
_From @raags on December 9, 2015 10:49_

Git module should have an option to run `git clean -f` to remove untracked files. This is useful to say build a project from a pristine repository. Currently all untracked files remain in the directory.

I think there should be two options :
- `clean_untracked` - remove files and directories
- `clean_ignored` - remove ignored flies

If this sounds good, I can send a PR

_Copied from original issue: ansible/ansible-modules-core#2656_

**Comment by benjifisher** [comment]
_From @shamdsc on January 20, 2016 11:14_

I second this.  I'm running `command: git clean -fd` at the moment before running `git: repo=...` so this would make it much neater.

**Comment by benjifisher** [comment]
_From @deyvsh on January 20, 2016 16:47_

Yep, I'd like to see this too and like the proposed implementation - thanks @raags.

**Comment by benjifisher** [comment]
_From @huiyiqun on February 2, 2016 10:7_

This is very useful for me.

**Comment by benjifisher** [comment]
_From @raags on April 20, 2016 17:41_

I have implemented this feature in PR with a better interface : https://github.com/ansible/ansible-modules-core/pull/2980 needs review

**Comment by benjifisher** [comment]
_From @ansibot on August 1, 2016 0:4_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by benjifisher** [comment]
_From @ansibot on September 9, 2016 19:57_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by benjifisher** [comment]
_From @ansibot on October 6, 2016 15:43_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by benjifisher** [comment]
_From @Kevinleaf on October 6, 2016 21:34_

+1 for me

**Comment by benjifisher** [comment]
_From @ansibot on October 22, 2016 9:7_

@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by benjifisher** [comment]
_From @ansibot on December 7, 2016 18:14_

This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
@benjifisher Greetings! Thanks for taking the time to open this issue. In order for the community to handle your issue effectively, we need a bit more information. 

Here are the items we could not find in your description:
- issue type
- ansible version
- component name

Please set the description of this issue with this template:
https://raw.githubusercontent.com/ansible/ansible/devel/.github/ISSUE_TEMPLATE.md

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibotdev** [comment]
@benjifisher You have not responded to information requests in this issue so we will assume it no longer affects you. If you are still interested in this, please create a new issue with the requested information.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by rjgonza** [comment]
Efforts on this die out?

**Comment by jwg4** [comment]
This should be reopened. The reason it has not been updated is that the OP is not the OP of the original issue on the other repo - this thread consists of two bots talking to each other.
